ts/a11y/sre.ts

```diff
@@ -16,51 +16,71 @@
  */
 
 /**
- * @fileoverview  Loads SRE (if necessary) and provides a promise for
- *                when it is ready.
+ * @fileoverview  Provides the interface functionality to SRE.
  *
  * @author dpvc@mathjax.org (Davide Cervone)
+ * @author v.sorge@mathjax.org (Volker Sorge)
  */
 
-import {asyncLoad} from '../util/AsyncLoad.js';
+import * as Api from 'speech-rule-engine/js/common/system.js';
+import {Walker} from 'speech-rule-engine/js/walker/walker.js';
+import * as WalkerFactory from 'speech-rule-engine/js/walker/walker_factory.js';
+import * as SpeechGeneratorFactory from 'speech-rule-engine/js/speech_generator/speech_generator_factory.js';
+import * as EngineConst from 'speech-rule-engine/js/common/engine_const.js';
+import Engine from 'speech-rule-engine/js/common/engine.js';
+import {ClearspeakPreferences} from 'speech-rule-engine/js/speech_rules/clearspeak_preferences.js';
+import {Highlighter} from 'speech-rule-engine/js/highlighter/highlighter.js';
+import * as HighlighterFactory from 'speech-rule-engine/js/highlighter/highlighter_factory.js';
+import {SpeechGenerator} from 'speech-rule-engine/js/speech_generator/speech_generator.js';
+import {Variables} from 'speech-rule-engine/js/common/variables.js';
+import MathMaps from './mathmaps.js';
 
-/**
- * The name of the modiule to load, if necessary
- */
-const SRELIB = (typeof window === 'undefined' ? './a11y/sre-node.js' :
-                '../../../speech-rule-engine/lib/sre_browser.js');
+export namespace Sre {
 
-/**
- * The promise for loading the SRE library, if it is not already loaded
- */
-const srePromise = (typeof sre === 'undefined' ? asyncLoad(SRELIB) : Promise.resolve());
+  export type highlighter = Highlighter;
 
-/**
- * Values to control the polling for when SRE is ready
- */
-const SRE_DELAY = 100;         // in milliseconds
-const SRE_TIMEOUT = 20 * 1000; // 20 seconds
+  export type speechGenerator = SpeechGenerator;
 
-/**
- * A promise that resolves when SRE is loaded and ready, and rejects if
- * SRE can't be loaded, or does not become ready within the timout period.
- */
-export const sreReady = function() {
```

Comment:
Since v3.2.1 is not supposed to be an update that includes breaking changes, I'm wondering if we shouldn't continue to export an `sreReady()` function that interfaces with the new promise structure of SRE, at least for the time being.  What do you think?